test/tools/d_do_test.d

```diff
@@ -419,7 +428,7 @@ string execute(ref File f, string command, bool expectpass, string result_path)
 
 string unifyNewLine(string str)
 {
-    static re = regex(`\r\n|\r|\n`, "g");
+    static re = regex(`\r\r\n|\r\n|\r|\n`, "g");
```

Comment:
@rainers - Looks like this is the only thing that works.

Reply:
Yes, see what all tests here have in common.  The original format string is something like:
```
Error: no identifier for declarator '%s'
```
The argument given is a multiline string.

Reply:
OK, thanks for tracking it down.  Should I leave a comment expressing such?

Perhaps the compiler should handle errors better in this regard though...

Reply:
A comment would be nice. 
IMO the compiler doesn't have to output any `\r' explicitly. It could use text mode when writing files if deemed necessary.